flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/JsonFileSystemFormatFactory.java

```diff
@@ -0,0 +1,242 @@
+		 */
+		private static final byte NEW_LINE = (byte) '\n';
+
+		private final DataType[] fieldTypes;
+		private final String[] fieldNames;
+		private final int[] selectFields;
+		private final List<String> partitionKeys;
+		private final String defaultPartValue;
+		private final long limit;
+		private final int[] jsonSelectFieldToProjectFieldMapping;
+		private final int[] jsonSelectFieldToJsonFieldMapping;
+		private final JsonRowDataDeserializationSchema deserializationSchema;
+
+		private transient boolean end;
+		private transient long emitted;
+		// reuse object for per record
+		private transient GenericRowData rowData;
+
+		public JsonInputFormat(
+				Path[] filePaths,
+				DataType[] fieldTypes,
+				String[] fieldNames,
+				int[] selectFields,
+				List<String> partitionKeys,
+				String defaultPartValue,
+				long limit,
+				int[] jsonSelectFieldToProjectFieldMapping,
+				int[] jsonSelectFieldToJsonFieldMapping,
+				JsonRowDataDeserializationSchema deserializationSchema) {
+			super.setFilePaths(filePaths);
+			this.fieldTypes = fieldTypes;
+			this.fieldNames = fieldNames;
+			this.selectFields = selectFields;
+			this.partitionKeys = partitionKeys;
+			this.defaultPartValue = defaultPartValue;
+			this.limit = limit;
+			this.jsonSelectFieldToProjectFieldMapping = jsonSelectFieldToProjectFieldMapping;
+			this.jsonSelectFieldToJsonFieldMapping = jsonSelectFieldToJsonFieldMapping;
+			this.deserializationSchema = deserializationSchema;
+		}
+
+		@Override
+		public boolean supportsMultiPaths() {
+			return true;
+		}
+
+		@Override
+		public void open(FileInputSplit split) throws IOException {
+			super.open(split);
+			this.end = false;
+			this.emitted = 0L;
+			this.rowData = PartitionPathUtils.fillPartitionValueForRecord(fieldNames, fieldTypes, selectFields,
+				partitionKeys, currentSplit.getPath(), defaultPartValue);
+		}
+
+		@Override
+		public boolean reachedEnd() {
+			return emitted >= limit || end;
+		}
+
+		@Override
+		public RowData readRecord(RowData reuse, byte[] bytes, int offset, int numBytes) throws IOException {
+			GenericRowData returnRecord = null;
+			do {
+				// remove \r from a line when the line ends with \r\n
+				if (this.getDelimiter() != null && this.getDelimiter().length == 1
+					&& this.getDelimiter()[0] == NEW_LINE && offset + numBytes >= 1
+					&& bytes[offset + numBytes - 1] == CARRIAGE_RETURN){
+					numBytes -= 1;
+				}
+				byte[] trimBytes = Arrays.copyOfRange(bytes, offset, offset + numBytes);
+				GenericRowData jsonRow = (GenericRowData) deserializationSchema.deserialize(trimBytes);
+
+				if (jsonRow != null) {
+					// if the record is null, simply just skip
+					returnRecord = rowData;
+					for (int i = 0; i < jsonSelectFieldToJsonFieldMapping.length; i++) {
+						returnRecord.setField(jsonSelectFieldToProjectFieldMapping[i],
+							jsonRow.getField(jsonSelectFieldToJsonFieldMapping[i]));
+					}
+				}
+			} while (returnRecord == null && !reachedEnd());
```

Comment:
```
RowData readRecord() {
...
  if (jsonRow != null) {
    return null;
  }
...
}

RowData nextRecord() {
   while (true) {
      if (readLine()) {
			RowData row = readRecord(record, this.currBuffer, this.currOffset, this.currLen);
             if (row == null) {
                 continue;
             } else {
                return row; 
            }
		} else {
			this.end = true;
			return null;
		}
  }
}
```

Reply:
so ,by this we can support ignore-parser-error and open by default?
that's a good tips